code/modules/atmospherics/machinery/components/unary_devices/cryo.dm

```diff
@@ -1,7 +1,58 @@
-#define CRYOMOBS 'icons/obj/cryo_mobs.dmi'
 ///Max temperature allowed inside the cryotube, should break before reaching this heat
 #define MAX_TEMPERATURE 4000
 
+/// This is a visual helper that shows the occupant inside the cryo cell.
+/atom/movable/visual/cryo_occupant
+	icon = 'icons/obj/cryogenics.dmi'
+	// Must be tall, otherwise the filter will consider this as a 32x32 tile
+	// and will crop the head off.
+	icon_state = "mask_bg"
+	layer = ABOVE_WINDOW_LAYER + 0.01
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	pixel_y = 22
+	appearance_flags = KEEP_TOGETHER
+	var/obj/machinery/atmospherics/components/unary/cryo_cell/cryo
+	var/animation = FALSE
+
+/atom/movable/visual/cryo_occupant/Initialize(mapload, obj/machinery/atmospherics/components/unary/cryo_cell/cryo_)
+	. = ..()
+	cryo = cryo_
+	// Alpha masking
+	// It will follow this as the animation goes, but that's no problem as the "mask" icon state
+	// already accounts for this.
+	filters += filter(type = "alpha", icon = icon('icons/obj/cryogenics.dmi', "mask"), y = -22)
+
+/atom/movable/visual/cryo_occupant/Destroy()
+	cryo = null
+	return ..()
+
+/atom/movable/visual/cryo_occupant/update_icon()
+	if(!cryo)
+		// Should only happen at init
+		return
+	if(animation)
+		if(!cryo.occupant || !cryo.on || !cryo.is_operational)
+			animate(src)
+			animation = FALSE
+		else
+			// Animation running: return early
+			return
+
+	// If there's no occupant, just make sure there are no vis_contents
+	if(!cryo.occupant)
+		vis_contents.Cut()
+		return
```

Comment:
Can you make the removal when the `occupant` is removed instead? React to that event?
That would make this code more expansible, instead of assuming there won't be other things to add to the `vis_contents` list ever.

Reply:
I'll change it but I would like to point out that this structure is supposed to represent the occupant only, so there shouldn't be more visual contents to it. Instead they should be added to the cryo.